src/transformStateWithClones.js

```diff
@@ -6,8 +6,32 @@
  *
  * @return {Object[]}
  */
-function transformStateWithClones(state, actions) {
-  // write code here
+function transformStateWithClones(initialState, actions) {
+  const stateHistory = [];
+  let currentState = { ...initialState }; // Клонування початкового стану
+
+  actions.forEach((action) => {
+    if (action.type === 'addProperties') {
+      // Додаємо властивості з extraData
+      currentState = { ...currentState, ...action.extraData };
```

Comment:
Ensure that `action.extraData` is defined and is an object before merging it with `currentState`. This will prevent potential runtime errors if `extraData` is undefined or not an object.